### PR TITLE
fixes for #339

### PR DESCRIPTION
**Pull request by mikewyszinski**
fixes for #339

### CODE REVIEW

File: packages/xchain-bitcoincash/src/utils.ts [hunk 1]
```diff
@@ -290,18 +285,23 @@ export const buildTx = async ({
...
-        const out = bitcash.address.toOutputScript(toLegacyAddress(output.address), bchNetwork(network))
-        transactionBuilder.addOutput(out, output.value)
+        out = bitcash.address.toOutputScript(toLegacyAddress(output.address), bchNetwork(network))
       }
+      transactionBuilder.addOutput(out, output.value)
     })
 
+    // add output for memo
+    if (compiledMemo) {
+      transactionBuilder.addOutput(compiledMemo, 0) // Add OP_RETURN {script, value}
```
Comment (stuartwk) [inline packages/xchain-bitcoincash/src/utils.ts:302]: is this adding the memo twice? I see on line 290 out is also being set as compiled memo?

**Review by thorswapdev — APPROVED** [review]
LGTM!

**Comment by thorswapdev** [comment]
@mikewyszinski , pls fix conflicts and let's get it published.

File: packages/xchain-bitcoincash/src/utils.ts (hunk 1 above)
Comment (mikewyszinski) [reply packages/xchain-bitcoincash/src/utils.ts:302]: thanks for catching this !

**Pull request merged**